{
  if (null == caughtException) {
    throw new IllegalArgumentException("Cannot obtain rootCause from (null)");
  }
  Throwable root=caughtException;
  Throwable cause=root.getCause();
  while (null != cause) {
    root=cause;
    cause=cause.getCause();
  }
  return root;
}
